src/System.Private.CoreLib/src/System/Reflection/Emit/DynamicILGenerator.cs

```diff
@@ -875,6 +875,20 @@ internal class DynamicILInfo
         private int m_methodSignature;
         #endregion
 
+        #region Constructor
+        internal DynamicILInfo(object scope, DynamicMethod method, byte[] methodSignature)
+        {
+            if (scope as DynamicScope is null)
+                throw new InvalidCastException(nameof(scope));
+            m_method = method;
```

Comment:
while you are here, you might wanna change the names of this variables to new convention i.e private fields starting with _

Reply:
I tried doing that, but there's a lot of these private fields with m_ in both DynamicILGenerator and DynamicILInfo. And since they already exist in code, I think it's better to leave as is.